response.go

```diff
@@ -0,0 +1,29 @@
+package sqlds
+
+import (
+	"sync"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+type Response struct {
+	res *backend.QueryDataResponse
+	mtx *sync.Mutex
+}
+
+func (r *Response) Set(refID string, res backend.DataResponse) {
+	r.mtx.Lock()
+	r.res.Responses[refID] = res
+	r.mtx.Unlock()
+}
+
+func (r *Response) Response() *backend.QueryDataResponse {
+	return r.res
```

Comment:
Here be dragons. Returning a pointer to memory we know is used concurrently can result in sadness. 

It might be better to deal entirely in values with this type and hold a pointer to the Response struct that only contains/returns values. We can let the caller take the pointer

Reply:
hmm. I think in this case it might be OK because we know when it's used (after the waitgroup is done), but that's definitely a good point and is something i'll go update.